cms/src/endpoints/preview/index.ts

```diff
@@ -0,0 +1,67 @@
+import { EndpointConfig, Accountability } from "@directus/shared/src/types"
+import axios from "axios"
+
+const apiPath = process.env.API_PATH;
+const apiSecret = process.env.API_SECRET;
+
+const getAccountability = (req: any) => {
+    return req.accountability as Accountability;
+}
+
+const getUserId = (req: any) => {
+    try {
+        return getAccountability(req).user ?? null;
+    } catch {
+        return null
+    }
+}
+
+const endpointConfig: EndpointConfig = {
+    id: "preview",
+    handler: (router) => {
+        router.post("/collection", async (req, res) => {
+            const userId = getUserId(req)
+            if (userId == null) {
+                res.status(401).send("Unauthenticated")
+                return
+            }
+            const filter = req.body.filter;
+            const collection = req.body.collection;
+
+            try {
+
+                const body = `query {
+                    preview {
+                        collection(collection: "${collection}", filter: ${JSON.stringify(JSON.stringify(filter))}) {
```

Comment:
Why do we need a double stringify? Is it basically an escape?
It would likely be best to write the queries with params here.

Reply:
First stringify: Convert object to json. Second stringify: convert json object to string.

The filter isn't typed, so this is the way I send the filter to the api for deserializing. Know of a better way?